compiler/rustc_resolve/src/lib.rs

```diff
@@ -999,7 +999,7 @@ pub struct Resolver<'a> {
 
     /// Table for mapping struct IDs into struct constructor IDs,
     /// it's not used during normal resolution, only for better error reporting.
-    struct_constructors: DefIdMap<(Res, ty::Visibility)>,
+    struct_constructors: DefIdMap<(Res, ty::Visibility, Vec<(ast::StructField, ty::Visibility)>)>,
```

Comment:
```suggestion
    struct_constructors: DefIdMap<(Res, Vec<(ty::Visibility, Span)>)>,
```
The more detailed field visibilities make the constructor visibility redundant, and we don't need the whole `ast::StructField`, only its span.

Reply:
Isn't that the span of the definition of the field? We want the span of the actual pattern, which is why I add the pattern idents to the pattern source

Reply:
Ah, right, so we can just use `Vec<ty::Visibility>`.